.github/workflows/ci.yml

```diff
@@ -0,0 +1,151 @@
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Conda Bootstrap
+        uses: goanpeca/setup-miniconda@v1
+        with:
+          auto-update-conda: true
+          channel-priority: strict
+          channels: conda-forge
+          python-version: ${{ matrix.python }}
+      - name: Conda Config
+        run: |
+          conda config --set always_yes yes
+          conda config --set changeps1 no
+          conda config --set pip_interop_enabled True
+
+      # Installation into the `test` environment
+      - name: Conda Install Pyarrow (non-nightly)
+        run: conda install pyarrow==${{ matrix.pyarrow }}
+        if: matrix.pyarrow != 'nightly'
+      - name: Conda Install Pyarrow (nightly)
+        # Install both arrow-cpp and pyarrow to make sure that we have the
+        # latest nightly of both packages. It is sadly not guaranteed that the
+        # nightlies and the latest release would otherwise work together.
+        run: conda install -c arrow-nightlies arrow-cpp pyarrow
+        if: matrix.pyarrow == 'nightly'
+      - name: Conda Install conda-requirements.txt + conda-test-requirements.txt
+        # run this AFTER pyarrow, since this is less restrictive (so conda doesn't spend too much time downgrading)
+        run: conda install --file=conda-requirements.txt --file=conda-test-requirements.txt
+      - name: Pip Instal NumFOCUS nightly
+        # NumFOCUS nightly wheels, contains numpy and pandas
+        run: env PRE_WHEELS="https://pypi.anaconda.org/scipy-wheels-nightly/simple" pip install --pre --upgrade --timeout=60 -f $PRE_WHEELS pandas numpy
+        if: matrix.numfocus_nightly
+      - name: Pip Install blacken-docs
+        # currently not on conda
+        run: pip install --no-deps blacken-docs
+      - name: Pip Install kartothek
+        run: pip install --no-deps -e .
+      - name: Conda Export
+        run: conda list --export
+
+      # Linters
+      - name: Flake8
+        run: flake8
+      - name: Mypy
+        run: mypy .
+      - name: Black
+        run: black --check .
+      - name: blacken-docs
+        run: blacken-docs
+      - name: Isort
+        run: isort --recursive --check-only
+
+      # Tests
+      - name: Pytest
+        run: pytest --cov kartothek --cov-report xml
+      - name: ASV
+        run: |
+          asv --config ./asv_bench/asv.conf.json machine --machine travis --os unknown --arch unknown --cpu unknown --ram unknown
+          asv --config ./asv_bench/asv.conf.json run --show-stderr --environment existing --quick
+      - name: Sphinx
+        run: |
+          python setup.py build_sphinx
+          touch ./docs/_build/html/.nojekyll
+        if: env.IS_MASTER_BUILD == 'true'
+
+      # Builds
+      - name: Build Wheel
+        run: python setup.py sdist bdist_wheel
+
+      # Artifacts
+      - name: Preserve Dist
+        uses: actions/upload-artifact@v1
+        with:
+          name: dist
+          path: dist
```

Comment:
do we want this for all jobs?

Reply:
The UI splits this on a per-job level, so they don't overlap. But they should be virtually identical, so we could change to to "master job only" (see `IS_MASTER_JOB`)

Reply:
I'm just trying to figure out what the value is in keeping all these distribution files